pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -0,0 +1,190 @@
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+)
+
+type DiffOptions struct {
+	FilenameOptions  resource.FilenameOptions
+	Mapper           meta.RESTMapper
+	Typer            runtime.ObjectTyper
+	Namespace        string
+	EnforceNamespace bool
+	DiffBuffer1      []byte
+	DiffBuffer2      []byte
+	Output           string
+	Codec            runtime.Encoder
+	Factory          cmdutil.Factory
+	Out              io.Writer
+	ErrOut           io.Writer
+	ext              string
+}
+
+func (o *DiffOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
+	if o.Output != "" {
+		if o.Output != "yaml" && o.Output != "json" {
+			return fmt.Errorf("invalid output format %s, only yaml|json supported", o.Output)
+		}
+	}
+
+	mapper, typer, err := f.UnstructuredObject()
+	if err != nil {
+		return err
+	}
+
+	o.Namespace, o.EnforceNamespace, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.Codec = runtime.NewCodec(f.JSONEncoder(), f.Decoder(true))
+	o.ext = "." + o.Output
+	o.Mapper = mapper
+	o.Typer = typer
+
+	r := resource.NewBuilder(o.Mapper, f.CategoryExpander(), o.Typer, resource.ClientMapperFunc(f.UnstructuredClientForMapping), unstructured.UnstructuredJSONScheme).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		FilenameParam(o.EnforceNamespace, &o.FilenameOptions).
+		Flatten().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		o.DiffBuffer1, err = runtime.Encode(o.Codec, info.VersionedObject)
+		if err != nil {
+			return err
+		}
+
+		// Verify the object exists in the cluster
+		if err := info.Get(); err != nil {
+			if apierrors.IsNotFound(err) {
+				return err
+			} else {
+				return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+			}
+		}
+
+		o.DiffBuffer2, err = kubectl.GetOriginalConfiguration(info.Mapping, info.Object)
+		if err != nil {
+			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+		}
+		if o.DiffBuffer2 == nil {
+			return cmdutil.UsageError(cmd, "no last-applied-configuration annotation found on resource: %s, to create the annotation, run the command with --create-annotation", info.Name)
+		}
+
+		//decode and delete the annotations fields
+		targetObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, o.DiffBuffer2)
+		if err != nil {
+			return nil
+		}
+		delete(targetObj.(runtime.Unstructured).UnstructuredContent()["metadata"].(map[string]interface{}), "annotations")
```

Comment:
I'm deleting the annotations in  [line16](https://github.com/kubernetes/kubernetes/blob/4597658cb9bf03640816b561c0ee036c4c1edcd5/pkg/kubectl/cmd/testdata/edit/testcase-apply-edit-last-applied-list/4.response#L16) instead of [line15](https://github.com/kubernetes/kubernetes/blob/4597658cb9bf03640816b561c0ee036c4c1edcd5/pkg/kubectl/cmd/testdata/edit/testcase-apply-edit-last-applied-list/4.response#L15), is that suppose to be always empty like this `\"annotations\":{},\"`?